src/Cli.Tests/ConfigGeneratorTests.cs

```diff
@@ -119,6 +121,77 @@ public void TestCleanup()
         }
     }
 
+    /// <summary>
+    /// Test to verify creation of initial config with special characters
+    /// such as [!,@,#,$,%,^,&,*, ,(,)] in connection-string or graphql api
+    /// </summary>
+    [TestMethod]
+    public void TestSpecialCharactersInConnectionString()
+    {
+        HandleConfigFileCreationAndDeletion(TEST_RUNTIME_CONFIG_FILE, configFilePresent: false);
+        InitOptions options = new(
+            databaseType: DatabaseType.MSSQL,
+            connectionString: "A!string@with#some$special%characters^to&check*proper(serialization).'",
+            cosmosNoSqlDatabase: null,
+            cosmosNoSqlContainer: null,
+            graphQLSchemaPath: null,
+            graphQLPath: "/An_",
+            setSessionContext: false,
+            hostMode: HostMode.Production,
+            corsOrigin: null,
+            authenticationProvider: EasyAuthType.StaticWebApps.ToString(),
+            config: TEST_RUNTIME_CONFIG_FILE);
+
+        StringBuilder expectedRuntimeConfigJson = new(
```

Comment:
Since this is a cherry pick, looks like it functionally works and the optimization for readability and re-usability can be considered for a future change.

Reply:
@aaronburtle, `StringBuilder` would be preferable to avoid creating multiple strings - because after each Replace, it will create a new string. This would save space.
Why do you think string.Replace is faster than StringBuilder?

Reply:
> @aaronburtle, `StringBuilder` would be preferable to avoid creating multiple strings - because after each Replace, it will create a new string. This would save space. Why do you think string.Replace is faster than StringBuilder?

The articles I found that did the most comprehensive testing find the `string.Replace` is a bit faster than `StringBuilder` but as you say can take up more memory.

https://learn.microsoft.com/en-us/archive/blogs/debuggingtoolbox/comparing-regex-replace-string-replace-and-stringbuilder-replace-which-has-better-performance

I actually just prefer the style of being able to call `.replace` again and again.

As to why it is faster, I think it has to do with the way that stringbuilder calls replace under the hood, and needing to build new strings there to do the replacement is slow.